nemo/collections/tts/losses/fastpitchloss.py

```diff
@@ -120,6 +120,37 @@ def forward(self, pitch_predicted, pitch_tgt, len):
         return pitch_loss
 
 
+class EnergyLoss(Loss):
+    def __init__(self, loss_scale=0.1):
+        super().__init__()
+        self.loss_scale = loss_scale
+
+    @property
+    def input_types(self):
+        return {
+            "energy_predicted": NeuralType(('B', 'T'), RegressionValuesType()),
+            "energy_tgt": NeuralType(('B', 'T'), RegressionValuesType()),
+            "len": NeuralType(('B'), LengthsType()),
+        }
+
+    @property
+    def output_types(self):
+        return {
+            "loss": NeuralType(elements_type=LossType()),
+        }
+
+    @typecheck()
+    def forward(self, energy_predicted, energy_tgt, len):
+        dur_mask = mask_from_lens(len, max_len=energy_tgt.size(1))
+        ldiff = energy_tgt.size(1) - energy_predicted.size(1)
+        energy_predicted = F.pad(energy_predicted, (0, ldiff, 0, 0), value=0.0)
```

Comment:
Also curious if this may run into the case where `ldiff` is negative.

Reply:
I suspect the reason this was added with the pitch loss is because librosa.pyin() has a bug in which it outputs 1 more audio frame that it is supposed to when the audio length is exactly divisible by the frame step size. But in that case, the correct action would be to remove the final audio from from the ground truth rather than pad a zero to the prediction.

In every other situation, this padding would indicate there is a larger bug.

Reply:
I think we can remove ldiff. I have tested with a small training run, and I have not noticed any difference.